### ISSUE TITLE
OffsetOperationsInspection + return new static()

### ISSUE THREAD

**Issue by lintaba** [kalessil/phpinspectionsea#1177]
First, thank you for reporting a bug or making a request. That takes time and we appreciate that!

Let's start with common details:
 
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-Positive                          |
| Plugin         | Php Inspections (EA Extended) |
| Language level | 7.1                       |
| Module | OffsetOperationsInspection |

Current behaviour (description/screenshot:)
--

```
$result = new Collection { public function keyBy($key) { return new static(); } };
$array = $result->keyBy("keys");
$key = "index";
$array[$index]; // <-- error here
```
Recognizes $array as `static`, instead of `Collection`
Collection: https://github.com/laravel/framework/blob/5.5/src/Illuminate/Support/Collection.php#L722
![image](https://user-images.githubusercontent.com/375078/51492401-f9455c00-1db1-11e9-8e68-21f9ed5db2ed.png)
> '$array' may not support offset operations (or its type not annotated properly: [static])

Expected behaviour
--
`static` is a special self-referring type, and should threated like so.

Environment details
---
PhpStorm 2018.3.3
Build #PS-183.5153.36, built on January 9, 2019
Licensed to PhpStorm Evaluator
Expiration date: February 16, 2019
JRE: 1.8.0_152-release-1343-b26 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.2

**Comment by kalessil** [comment]
Thank you for reporting @lintaba, I'll take a look why this gets reported. Fixing might take a while due to high bug-reports rate.

**Comment by kalessil** [comment]
Fixed!
